spring-cloud-data-module-deployers/spring-cloud-data-module-deployer-cloudfoundry/src/main/java/org/springframework/cloud/data/module/deployer/cloudfoundry/CloudControllerTemplate.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.data.module.deployer.cloudfoundry;
+
+import java.net.URI;
+import java.util.ArrayList;
+
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.util.StringUtils;
+import org.springframework.web.util.UriComponentsBuilder;
+
+/**
+ * REST client specialised for module deployment requirements.
+ *
+ * @author Steve Powell
+ * @author Paul Harris
+ * @author Eric Bottard
+ */
+class CloudControllerTemplate implements CloudControllerOperations {
+
+	private final URI endpoint;
+
+	private final ExtendedOAuth2RestOperations restOperations;
+
+	CloudControllerTemplate(URI endpoint, ExtendedOAuth2RestOperations restOperations) {
+		this.endpoint = endpoint;
+		this.restOperations = restOperations;
+	}
+
+	@Override
+	public ListRoutesResponse listRoutes(ListRoutesRequest request) {
+		URI uri = UriComponentsBuilder.fromUri(this.endpoint)
+				.pathSegment("v2", "routes")
```

Comment:
We can possibly create a constant for `v2` as we know we are using CloudController v2 REST API. And, may be we can have this constant at `CloudControllerOperations` ?